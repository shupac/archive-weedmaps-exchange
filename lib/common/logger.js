const Logger = require('js-logger');
const serializeError = require('serialize-error');

const env = process.env.NODE_ENV || 'development';
const defaultLevel = process.env.LOG_LEVEL
  ? Logger[process.env.LOG_LEVEL.toUpperCase()]
  : Logger.DEBUG;

// Set the log level to what the ENV is set to or debug
Logger.useDefaults({
  defaultLevel,
  formatter: (messages, context) => {
    let args = false;
    if (messages.length > 1) {
      args = messages.splice(1);
    }
    const log = {
      message: messages[0],
      component: context.name
        ? context.name.replace('@ghostgroup/admin:', 'APP:')
        : '-',
      data: args === false ? undefined : args,
      log_level: context.level.name,
      timestamp: new Date().toUTCString(),
    };
    if (env !== 'development') {
      try {
        messages[0] = JSON.stringify(log);
      } catch (e) {
        log.message = serializeError(log.message);
        log.data = serializeError(log.data);
        messages[0] = JSON.stringify(log);
      }
    } else {
      messages[0] = `[${log.timestamp}] <${log.log_level}> (${
        log.component
      }): `;
      if (typeof log.message === 'string') {
        messages[0] = `${messages[0]} ${log.message}`;
        if (args) {
          messages[1] = log.data;
        }
      } else {
        messages[1] = log.message;
        if (args) {
          messages[2] = log.data;
        }
      }
    }
  },
});

module.exports = Logger;
