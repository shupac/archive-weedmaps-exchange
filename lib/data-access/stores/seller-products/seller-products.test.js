// @flow
import mockProductsResponse from 'mocks/seller-products';
import mockProductDetails, {
  mockErrorProduct,
  mockErrors,
} from 'mocks/seller-product-details';
import logger from 'lib/common/logger';

import SellerProducts from './';

function setup(mockRes) {
  const mockClient = {
    fetch: jest.fn().mockReturnValue(mockRes),
    patch: jest.fn().mockReturnValue(mockRes),
  };
  const sellerProducts = SellerProducts.create({}, { client: mockClient });
  return { sellerProducts, mockClient };
}

describe('SellerProducts Store', () => {
  it('should fetch seller products and set data', async () => {
    const { sellerProducts } = setup(mockProductsResponse);
    await sellerProducts.fetchProducts({});
    expect(sellerProducts.sellerProducts).toMatchSnapshot();
  });

  it('should transform query params into query string', () => {
    const { sellerProducts, mockClient } = setup(mockProductsResponse);
    sellerProducts.fetchProducts({
      active: 'published',
      categories: '1/2/3',
      page: 2,
      pageSize: 15,
      order: '-name',
    });

    expect(mockClient.fetch).toHaveBeenCalledWith(
      '/seller/products?filter%5Bactive%5D=true&filter%5Bcategory_ids%5D[]=1&filter%5Bcategory_ids%5D[]=2&filter%5Bcategory_ids%5D[]=3&page[number]=2&page[size]=15',
    );
  });

  it('should fetch product details and set data', async () => {
    const { sellerProducts } = setup({ data: mockProductDetails });
    await sellerProducts.fetchProductDetails('123');
    expect(sellerProducts.sellerProductDetails).toMatchSnapshot();
  });

  it('should update seller product', async () => {
    const updatedProduct = {
      ...mockProductsResponse.data[0],
      active: true,
    };
    const { sellerProducts } = setup({ data: updatedProduct });
    sellerProducts.setSellerProducts(mockProductsResponse.data);
    await sellerProducts.updateSellerProduct(updatedProduct);
    expect(sellerProducts.sellerProductDetails.active).toEqual(true);
    expect(
      sellerProducts.sellerProducts.find(
        ({ id }) => id === mockProductsResponse.data[0].id,
      ).active,
    ).toEqual(true);
  });

  it('should catch errors', () => {
    const mockErrorClient = {
      fetch: () => {
        throw new Error('test');
      },
    };

    const sellerProducts = SellerProducts.create(
      {},
      {
        client: mockErrorClient,
      },
    );
    const logError = jest.spyOn(logger, 'error').mockReturnValue();

    sellerProducts.fetchProducts({});
    expect(logError).toHaveBeenCalledWith(new Error('test'));
    sellerProducts.fetchProductDetails();
    expect(logError).toHaveBeenCalledWith(new Error('test'));
    logError.mockRestore();
  });

  it('should catch form errors', async done => {
    const mockErrorClient = {
      patch: () => {
        const error = new Error('test');
        throw error;
      },
    };

    const sellerProducts = SellerProducts.create(
      {},
      {
        client: mockErrorClient,
      },
    );

    const response = await sellerProducts.updateSellerProduct(mockErrorProduct);
    expect(response).toEqual(false);
    done();
  });

  it('should catch and set the form errors if available', async done => {
    const mockErrorClient = {
      patch: () => {
        const error = new Error('test');
        // $FlowFixMe
        error.response = {
          json: () => Promise.resolve({ errors: mockErrors }),
        };
        throw error;
      },
    };

    const sellerProducts = SellerProducts.create(
      {},
      {
        client: mockErrorClient,
      },
    );

    const setDetailFormErrors = jest.spyOn(
      sellerProducts,
      'setDetailFormErrors',
    );
    const response = await sellerProducts.updateSellerProduct(mockErrorProduct);
    expect(setDetailFormErrors).toHaveBeenCalledWith(mockErrors);
    expect(sellerProducts.productFormErrors).toEqual(mockErrors);
    expect(response).toEqual(false);
    done();
  });
});
