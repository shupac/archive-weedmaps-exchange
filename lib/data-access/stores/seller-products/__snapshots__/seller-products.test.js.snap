// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerProducts Store should fetch product details and set data 1`] = `
Object {
  "active": true,
  "categoryNames": Object {
    "categoryName": "Gummies",
    "departmentName": "Edibles",
  },
  "id": "05f56e24-5842-4d0a-a4f0-ad18a41e2f0b",
  "product": Object {
    "avatarImage": Object {
      "id": "922327d6-1ad8-49d4-b9d6-3455d950a8f2",
      "largeUrl": "https://images-acceptance.weedmaps.com/products/000/054/200/avatar/large/1507933286-THCLEAR-GUMMI-BLUEB-250.png",
      "mediumUrl": "https://images-acceptance.weedmaps.com/products/000/054/200/avatar/medium/1507933286-THCLEAR-GUMMI-BLUEB-250.png",
      "smallUrl": "https://images-acceptance.weedmaps.com/products/000/054/200/avatar/square/1507933286-THCLEAR-GUMMI-BLUEB-250.png",
    },
    "brand": Object {
      "avatarImage": Object {
        "id": "202793d4-1967-432f-bbf1-6a21842f3abf",
      },
      "deliveryEta": Object {
        "etaMax": 3,
        "etaMaxUnit": "day",
        "etaMin": 1,
        "etaMinUnit": "day",
      },
      "description": "<p>We are the industry leader in solventless distillate. We currently produce 6 different products with our Thclear Distillate, with our infamous uncut Private Reserve Cartridge being the most popular. Our other amazing products include: Disposable Pens, Cartridges, Syringes, Honeypot Vials, and our funky 50/50!</p>",
      "id": "d60ca06a-89db-4584-99a8-1f9b9f427e42",
      "licenses": Array [],
      "minimumPurchasePrice": 10000,
      "name": "THClear Co",
      "shippingFee": 100,
      "slug": "thclear-co",
    },
    "categories": Array [
      Object {
        "id": "af3eeab3-26f9-425b-99b5-fa70021e0d45",
        "name": "Gummies",
        "slug": "gummies",
      },
    ],
    "departments": Array [
      Object {
        "id": "e306a27d-5076-4074-ab7a-21f6485ad4bb",
        "name": "Edibles",
        "slug": "edibles",
      },
    ],
    "description": "<p>THClear blueberry gummies is made with lots of high-grade THC that delivers an unbelievably tasty flavor that leaves a blueberry aroma followed by strong feelings of happiness, creativity and euphoria. Many edible-lovers enjoy this tasty treat for its wonderful natural effects that aid in relieving stress, pain and anxiety.</p>",
    "galleryImages": Array [
      Object {
        "id": "9000aaef-48c2-4691-bcfd-d6af94c66ece",
        "largeUrl": "https://images-acceptance.weedmaps.com/photos/products/000/054/200/large/104972_THCLEAR-GUMMI-BLUEB-250.png",
        "mediumUrl": "https://images-acceptance.weedmaps.com/photos/products/000/054/200/medium/104972_THCLEAR-GUMMI-BLUEB-250.png",
        "smallUrl": "https://images-acceptance.weedmaps.com/photos/products/000/054/200/square/104972_THCLEAR-GUMMI-BLUEB-250.png",
      },
    ],
    "id": "05f56e24-5842-4d0a-a4f0-ad18a41e2f0b",
    "name": "BLUEBERRY GUMMIES - 250 MG",
    "slug": "thclear-co-blueberry-gummies-250-mg",
    "variants": Array [
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 75,
            "currency": "usd",
            "id": "fe745d99-d90d-41fe-888e-153601d39a90",
            "price": "60",
          },
        ],
        "id": "91d83169-d619-483d-92ff-3fb1492bebb5",
        "name": "1 G Test Acc(DELETE WHEN DONE)",
        "size": 1,
        "sku": "onegee",
        "unit": "gram",
      },
      Object {
        "allocations": Array [],
        "id": "04f16514-29fd-4835-aa91-4fbf3e3b28e3",
        "name": "Gram",
        "size": 2,
        "sku": "2gram2gram2gram2gram",
        "unit": "gram",
      },
      Object {
        "allocations": Array [],
        "id": "bd9a0d83-10ad-4689-afab-16d0fb1ad802",
        "name": "Gram",
        "size": 3,
        "sku": "3gram3gram3gram3gram",
        "unit": "gram",
      },
      Object {
        "allocations": Array [],
        "id": "692b6845-bd9d-42e6-8672-d459e1c8d4b2",
        "name": "Gram",
        "size": 4,
        "sku": "4gram4gram4gram4gram",
        "unit": "gram",
      },
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 5,
            "currency": "usd",
            "id": "10a519bf-bc94-4d73-95c9-532080b52ade",
            "price": "150",
          },
          Object {
            "active": true,
            "amount": 0,
            "currency": "usd",
            "id": "dfaaa1b3-b3c4-4031-b475-ca8c77919c13",
            "price": "150",
          },
        ],
        "id": "e0c6412f-05e4-4779-b964-86932d4de96a",
        "name": "Pack of 5",
        "size": 5,
        "sku": "15555555555",
        "unit": "gram",
      },
      Object {
        "allocations": Array [],
        "id": "1b9d74fd-e74a-4b7d-acf1-be7a32394ab0",
        "name": "Gram",
        "size": 5,
        "sku": "5gram5gram5gram5gram",
        "unit": "gram",
      },
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 10,
            "currency": "usd",
            "id": "c4e8a036-da84-4a44-9502-fbd6593dd0b3",
            "price": "290",
          },
          Object {
            "active": true,
            "amount": 1999,
            "currency": "usd",
            "id": "fbc69d33-c669-412b-8743-5ff84c07ff0d",
            "price": "500",
          },
        ],
        "id": "1e64b030-2f47-4f64-a3fa-183368d66a22",
        "name": "Pack of 10",
        "size": 10,
        "sku": "1010",
        "unit": "gram",
      },
      Object {
        "allocations": Array [],
        "id": "8bba5489-5166-4af1-bfa8-4f5603d2d8c5",
        "name": "Gram",
        "size": 10,
        "sku": "10gram10gram10gram10gram",
        "unit": "gram",
      },
      Object {
        "allocations": Array [],
        "id": "c6ec5ee6-9d6e-491f-8b29-3c980d8b5745",
        "name": "Gram",
        "size": 20,
        "sku": "20gram20gram20gram20gram",
        "unit": "gram",
      },
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 10,
            "currency": "usd",
            "id": "e3ac1ecf-99fe-486f-8691-4eac6c88350a",
            "price": "580",
          },
          Object {
            "active": true,
            "amount": 9,
            "currency": "usd",
            "id": "60bf4a4e-a611-4b69-a73d-604036eab6ef",
            "price": "580",
          },
        ],
        "id": "fc393e48-aad7-4b66-b9ab-2dd2a4c99280",
        "name": "Pack of 20",
        "size": 20,
        "sku": "20202020220",
        "unit": "gram",
      },
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 75,
            "currency": "usd",
            "id": "d7a4da2f-ba0b-4d36-afba-5463d451c723",
            "price": "600",
          },
        ],
        "id": "e67c8429-5614-449b-96a2-7ecb15755b5f",
        "name": "Liter Variant",
        "size": 1,
        "sku": "onegee",
        "unit": "liter",
      },
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 437,
            "currency": "usd",
            "id": "2b7619ef-b77f-40d7-a04c-73f1f0078605",
            "price": "1000",
          },
        ],
        "id": "6c149b1d-86de-4178-8b9e-2b5658de884d",
        "name": "50 Units",
        "size": 50,
        "sku": "zzzzzzzz",
        "unit": "unit",
      },
      Object {
        "allocations": Array [],
        "id": "b6e35903-26dd-4919-8c92-2bfa1bceec3f",
        "name": "100 Units",
        "size": 100,
        "sku": "zzzzzzzz",
        "unit": "unit",
      },
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 75000,
            "currency": "usd",
            "id": "50ea6bab-524e-4e05-8ae9-14d3fdd751eb",
            "price": "1700",
          },
        ],
        "id": "c8143613-9717-4ea9-9b03-a5a3b1dd5095",
        "name": "Gallon Variant",
        "size": 1,
        "sku": "onegee",
        "unit": "gallon",
      },
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 496,
            "currency": "usd",
            "id": "7e4f45f5-e60a-4faa-aa48-0ee86f051066",
            "price": "40000",
          },
        ],
        "id": "597a1148-29ce-4608-bc67-8b863a76c7c4",
        "name": "500 Units",
        "size": 500,
        "sku": "zzzzzzzz",
        "unit": "unit",
      },
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 491,
            "currency": "usd",
            "id": "fff86083-41d4-447b-bfe0-459e2759ab3b",
            "price": "55000",
          },
        ],
        "id": "b8befbbe-433e-4eac-88c9-f7b539400fc2",
        "name": "1000 Units",
        "size": 1000,
        "sku": "zzzzzzzz",
        "unit": "unit",
      },
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 491,
            "currency": "usd",
            "id": "254dae57-b528-4803-bbf5-365080273638",
            "price": "106000",
          },
        ],
        "id": "b531f91c-fe94-4c9a-8ace-018b722a38e9",
        "name": "1500 Units",
        "size": 1500,
        "sku": "zzzzzzzz",
        "unit": "unit",
      },
    ],
  },
  "quantity": 79098,
  "variantCount": 17,
  "zoneCount": 13,
}
`;

exports[`SellerProducts Store should fetch seller products and set data 1`] = `
Array [
  Object {
    "active": false,
    "categoryNames": Object {
      "categoryName": "Live Resin",
      "departmentName": "Concentrates",
    },
    "id": "b3f4fd48-6698-43cd-addc-33504fcd0434",
    "product": Object {
      "avatarImage": Object {
        "id": "4dabb56e-4078-4f54-ab43-9b4fc5553c1e",
        "largeUrl": "https://images-acceptance.weedmaps.com/products/000/024/998/avatar/large/natureslab_live24k-1.jpg",
        "mediumUrl": "https://images-acceptance.weedmaps.com/products/000/024/998/avatar/medium/natureslab_live24k-1.jpg",
        "smallUrl": "https://images-acceptance.weedmaps.com/products/000/024/998/avatar/square/natureslab_live24k-1.jpg",
      },
      "brand": Object {
        "avatarImage": Object {
          "id": "ae387ff7-73b7-442c-b3bd-5221bb342bf4",
        },
        "deliveryEta": Object {
          "etaMax": 3,
          "etaMaxUnit": "day",
          "etaMin": 1,
          "etaMinUnit": "day",
        },
        "description": "<p><strong><ins>Our Mission:</ins></strong></p> <p><em>Do it for the Lab.</em></p> <p> </p> <p>Nature's Lab Extracts is hardly your average extract company. They specialize in creating honest products, crafted from the highest quality ingredients and cutting edge techniques to bring quality medicine to patients in California. Prioritizing greatness, they strive to offer connoisseur grade cannabis goods to the discerning consumer that are ripe with flavor, highly potent, and consistently excellent. Nature's Lab Extracts offers a happy, satisfied dabbing experience. Every time.</p> <p><strong><ins>History:</ins></strong></p> <p>While it didn’t start out with the name until later, Nature’s Lab Extracts began in the marijuana industry stretching back to 2008, where the founder transitioned from a background in mechanical engineering to cannabis extraction. Seeing the unsafe methods utilized back in the early days of concentrates, he knew their was a safer way to extract cannabis without the risk of blowing something up. Perfecting his BHO extraction methods, the founder began Nature’s Lab Extracts to share his creations with the marijuana world. Years later Nature’s Lab Extracts has grown an immense following, available in nearly 200 dispensaries throughout California and winning forty-five awards to date for their extractions.</p> <p><strong><ins>Process:</ins></strong></p> <p>Nature’s Lab Extracts utilizes the same extraction methods that have worked since the beginning while continually innovating for better ways to create their products. Using BHO methods, they extract their live resin and Nug Run concentrates from their in-house grown cannabis, cultivating strains their patients want instead of random genetics. They’ve exacted the science of their extractions to offer 0 parts per million purity from their high-quality flowers, and their Spacewalker strain (a Skywalker OG backcross) and G13 (an OG cross) indicate their dedication to excellence.</p> <p><strong><ins>Products:</ins></strong></p> <p>Products: A connoisseur's company, Nature’s Lab Extracts specializes in crafting Live Resins and Nug Run extractions; all pulled from organically cultivated and incredibly potent cannabis flower. Patients love their many strains, including their in-house Spacewalker, their Spacey, and their C-4. As they expand, Nature’s Lab Extracts plans on launching a vape pen line with unique terpene profiles down the line.</p> <p><strong><ins>Lab Testing:</ins></strong></p> <p>Nature’s Lab Extracts proudly maintains zero parts per million contaminants in their extracts, and tests with Anresco Laboratories in San Francisco to verify their record. They test each batch for pesticides, residual solvents, bacteria, and potency.</p> <p><strong><ins>Networks:</ins></strong></p> <p>Nature’s Lab Extracts is proud to line the shelves of 160 stores throughout the state of California. Patients can find their products in San Francisco, Sacramento, Bakersfield, Orange County, Los Angeles, the Inland Empire and San Diego. Notable dispensaries include Green Wolf, Neighborhood Wellness Center and Harborside, and they collaborate with Cookies, Russian Assassin Boys, Nature’s Farm, DNA Genetics, Nameless Genetics and Bert Baccarat to create exciting unique cannabis products.</p>",
        "id": "79577912-8b01-4194-b69d-e988a91d5113",
        "licenses": Array [],
        "minimumPurchasePrice": 0,
        "name": "Natures Lab Extracts",
        "shippingFee": 0,
        "slug": "natures-lab-extracts",
      },
      "categories": Array [
        Object {
          "id": "b7295924-975c-4e0c-bb18-250d4b233a35",
          "name": "Live Resin",
          "slug": "live-resin",
        },
      ],
      "departments": Array [
        Object {
          "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
          "name": "Concentrates",
          "slug": "concentrates",
        },
      ],
      "description": "<p>Nature’s Lab Extracts 24K Live Resin Shatter has strong citrusy-sweet, orange flavor and effects that put users into a cheery mood. 24K’s effects leave users with a relaxation that builds without inducing laziness. While the Sativa part of this Indica dominant shatter introduces a light amount of energy.</p>",
      "galleryImages": Array [
        Object {
          "id": "68923893-28b4-4641-8223-eebd4f69f5a8",
          "largeUrl": "https://images-acceptance.weedmaps.com/photos/products/000/024/998/large/33894_natureslab_live24k-2.jpg",
          "mediumUrl": "https://images-acceptance.weedmaps.com/photos/products/000/024/998/medium/33894_natureslab_live24k-2.jpg",
          "smallUrl": "https://images-acceptance.weedmaps.com/photos/products/000/024/998/square/33894_natureslab_live24k-2.jpg",
        },
        Object {
          "id": "8e67e4d3-7e29-425f-b897-67c0ee0c1d31",
          "largeUrl": "https://images-acceptance.weedmaps.com/photos/products/000/024/998/large/33944_natureslab_live24k-1.jpg",
          "mediumUrl": "https://images-acceptance.weedmaps.com/photos/products/000/024/998/medium/33944_natureslab_live24k-1.jpg",
          "smallUrl": "https://images-acceptance.weedmaps.com/photos/products/000/024/998/square/33944_natureslab_live24k-1.jpg",
        },
      ],
      "id": "b3f4fd48-6698-43cd-addc-33504fcd0434",
      "name": "24K Live Resin Shatter",
      "slug": "nature-s-lab-extracts-24k-live-resin-shatter",
      "variants": Array [],
    },
    "quantity": 5,
    "variantCount": 2,
    "zoneCount": 0,
  },
  Object {
    "active": false,
    "categoryNames": Object {
      "categoryName": "Men's Apparel",
      "departmentName": "Gear",
    },
    "id": "914a63ff-a67b-47fb-946b-86c991ce4560",
    "product": Object {
      "avatarImage": Object {
        "id": "a33aae83-fa32-470f-9ddb-940b704a3a2f",
        "largeUrl": "https://images-acceptance.weedmaps.com/products/000/016/381/avatar/large/blvdsupply_2_palmstee_1.png",
        "mediumUrl": "https://images-acceptance.weedmaps.com/products/000/016/381/avatar/medium/blvdsupply_2_palmstee_1.png",
        "smallUrl": "https://images-acceptance.weedmaps.com/products/000/016/381/avatar/square/blvdsupply_2_palmstee_1.png",
      },
      "brand": Object {
        "avatarImage": undefined,
        "deliveryEta": Object {
          "etaMax": 3,
          "etaMaxUnit": "day",
          "etaMin": 1,
          "etaMinUnit": "day",
        },
        "description": "",
        "id": "9ffabab9-75bd-4d17-b8f6-265470243155",
        "licenses": Array [],
        "minimumPurchasePrice": 0,
        "name": "BLVD Supply",
        "shippingFee": 0,
        "slug": "blvd-supply",
      },
      "categories": Array [
        Object {
          "id": "bbedabf1-a8c3-4fcb-830b-3b053a706b8f",
          "name": "Men's Apparel",
          "slug": "men-s-apparel",
        },
      ],
      "departments": Array [
        Object {
          "id": "4a257a1c-7f54-4be3-ba6f-53fde73fb5f6",
          "name": "Gear",
          "slug": "gear",
        },
      ],
      "description": "<p>BLVD Supply's 2 Palms short sleeve tee has an eyecatching focal point, two crossed AK-47's, sprouting two unassuming palms. It also has our rare \\"From the Sand to the Streets\\" tagline which, in this instance make more sense to be reversed, \\"From the Streets to the Sand.\\"  Available in Sm, Med, Lg, Xl, XXl and XXXl. Premium quality Cotton blend. Runs true to size. Machine wash cold with similar colors.</p>",
      "galleryImages": Array [
        Object {
          "id": "bcd19d62-d1c2-46f2-89b3-4561238d2d47",
          "largeUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/large/16463_blvdsupply_2_palmstee_2.png",
          "mediumUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/medium/16463_blvdsupply_2_palmstee_2.png",
          "smallUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/square/16463_blvdsupply_2_palmstee_2.png",
        },
        Object {
          "id": "d21838ea-233a-46ea-9ecf-6cd548057bcd",
          "largeUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/large/16359_blvdsupply_2_palmstee_1.png",
          "mediumUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/medium/16359_blvdsupply_2_palmstee_1.png",
          "smallUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/square/16359_blvdsupply_2_palmstee_1.png",
        },
        Object {
          "id": "d861f2b8-6d88-41b4-918d-ee42c6c03c1d",
          "largeUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/large/16532_blvdsupply_2_palmstee_3.png",
          "mediumUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/medium/16532_blvdsupply_2_palmstee_3.png",
          "smallUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/square/16532_blvdsupply_2_palmstee_3.png",
        },
        Object {
          "id": "f890d2f4-e2ca-4041-b88e-95732cc8b674",
          "largeUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/large/16399_blvdsupply_2_palmstee_4.png",
          "mediumUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/medium/16399_blvdsupply_2_palmstee_4.png",
          "smallUrl": "https://images-acceptance.weedmaps.com/photos/products/000/016/381/square/16399_blvdsupply_2_palmstee_4.png",
        },
      ],
      "id": "914a63ff-a67b-47fb-946b-86c991ce4560",
      "name": "2 Palms Tee",
      "slug": "blvd-supply-2-palms-tee",
      "variants": Array [],
    },
    "quantity": 5,
    "variantCount": 1,
    "zoneCount": 0,
  },
  Object {
    "active": true,
    "categoryNames": Object {
      "categoryName": "",
      "departmentName": "Concentrates",
    },
    "id": "554cc78e-d9f3-4240-8b57-63a377c5a240",
    "product": Object {
      "avatarImage": Object {
        "id": "c539e6a9-ae31-4835-8c03-71c71c7883a1",
        "largeUrl": "https://images.weedmaps.com/products/000/052/107/avatar/large/1506462031-170918_303KushLW_3874.jpg?development=1",
        "mediumUrl": "https://images.weedmaps.com/products/000/052/107/avatar/medium/1506462031-170918_303KushLW_3874.jpg?development=1",
        "smallUrl": "https://images.weedmaps.com/products/000/052/107/avatar/square/1506462031-170918_303KushLW_3874.jpg?development=1",
      },
      "brand": Object {
        "avatarImage": Object {
          "id": "f8754a5a-01bd-46b3-ac72-5d2ae311ad28",
        },
        "deliveryEta": Object {
          "etaMax": 3,
          "etaMaxUnit": "day",
          "etaMin": 1,
          "etaMinUnit": "day",
        },
        "description": "<p><ins><strong>Our Mission:</strong></ins></p> <p><em>Get Harmonized.</em></p> <p> </p> <p>Harmony Extracts passionately champions the cannabis and those within it. Wanting to do their part for the cannabis community, Harmony Extracts develops invariably excellent extract products in a long-time heart of the industry: Denver, Colorado. Here for everyone from the average smoker to the connoisseur market, Harmony Extracts assembles a team from the Old Guard of cannabis to bring the best quality and cleanest product to market at the most affordable prices. At Harmony Extracts, excellence is their norm.</p> <p><strong><ins>History:</ins></strong></p> <p>While Harmony Extracts officially started in August 2016, the passionate team behind the brand holds more than a decade’s worth of experience in the cannabis industry. Starting in 2003, the partners of Harmony Extracts began in cultivation before expanding into extraction and other leadership positions as the Colorado marijuana market became first medical and then recreational. As this team did everything from run cultivation centers to operate dispensaries, they soon felt a void in the marijuana market for consistent and quality extract product. The founders wanted to do things the right way, improving things from both a customer service and product standpoint and introducing a professionalism to the cannabis industry. Thus, Harmony Extracts sprung onto the scene fighting tooth and nail to meet Denver’s strict quality regulations and offer consistent, reliably excellent extractions.</p> <p>Today, Harmony Extracts lines the shelves of thirty dispensaries, where patients and recreational users alike notice their quality and recognize their brand. Harmony Extracts produces their world class products from their two state-of-the-art extraction facilities in Denver.</p> <p><strong><ins>Process:</ins></strong></p> <p>Hailing from many larger names in the industry, the extractors behind Harmony Extracts joined forces and combined the best methods to create their cannabis product. All of their flowers they grow in-house, or purchase from one or two of their exclusive vendors, to ensure their best materials remain the best. Drawing from classic strains and new, exciting genetics, Harmony Extracts uses butane and propane proprietary blend extraction to create the clearest and most potent shatter and wax. Their flagship product, their “Nectar” offers users an ultra-refined hydrocarbon extract with the potency of a distillate and the flavor of a live resin. Their proprietary method of extraction allows them to create a terpene-rich solution that crystallizes THCA, enabling them to separate the high-terpene cannabinoids from the crystals and reintroduce the plant materials at the end.</p> <p><strong><ins>Products:</ins></strong></p> <p>Harmony Extracts’ products started at “good” and have grown into “great” today.hey continually reevaluate their methods to keep delivering consistent excellence to their consumers. At the 2016 Hemp Connoisseur Championship, their four entries brought home eight different awards for Best Live Resin, Shatter and Wax. Always promising a high potency, their Nectar regularly tests at a potency of 99.99% THCA</p> <p><strong><ins>Lab Testing:</ins></strong></p> <p>Harmony Extracts exclusively tests their products with TEQ Analytical Laboratories in Aurora, Colorado, double checking their extracts for potency, terpene content, and any residual solvents.</p> <p><strong><ins>Networks:</ins></strong></p> <p>Currently, Harmony Extracts distributes their concentrate products to dispensaries throughout Colorado, found in popular dispensaries including Kind Love, Good Chemistry, Colorado Harvest Company and DANK in Denver, Boulder, and Crescent View. While they currently only work with recreational stores, they soon will offer products to medical patients. Harmony Extracts hopes to expand to more states by the start of 2018.</p>",
        "id": "e98a5787-2e60-4302-b566-c6454a69a91f",
        "licenses": Array [],
        "minimumPurchasePrice": 0,
        "name": "Harmony Extracts",
        "shippingFee": 0,
        "slug": "harmony-extracts",
      },
      "categories": Array [
        Object {
          "id": "6c56279f-be48-44be-8a53-8fb8303e154c",
          "name": "Budder",
          "slug": "budder",
        },
        Object {
          "id": "b7295924-975c-4e0c-bb18-250d4b233a35",
          "name": "Live Resin",
          "slug": "live-resin",
        },
      ],
      "departments": Array [
        Object {
          "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
          "name": "Concentrates",
          "slug": "concentrates",
        },
        Object {
          "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
          "name": "Concentrates",
          "slug": "concentrates",
        },
      ],
      "description": "<p>Our waxes are made to perform. We start with high-THC plant strains, extract all the goodness nature has to offer, and whip in some O2 to create the perfect consistency. Depending on the source material, the end product can either be a more dry crumbly wax, or a soft thick budder. Waxes and budders are known for their great flavor and aroma, along with being very user-friendly.</p>",
      "galleryImages": Array [
        Object {
          "id": "a1816cde-4d9b-4066-96ee-504ca269b197",
          "largeUrl": "https://images.weedmaps.com/photos/products/000/052/107/large/99970_170918_303KushLW_3874.jpg?development=1",
          "mediumUrl": "https://images.weedmaps.com/photos/products/000/052/107/medium/99970_170918_303KushLW_3874.jpg?development=1",
          "smallUrl": "https://images.weedmaps.com/photos/products/000/052/107/square/99970_170918_303KushLW_3874.jpg?development=1",
        },
      ],
      "id": "554cc78e-d9f3-4240-8b57-63a377c5a240",
      "name": "303 Kush Live Wax",
      "slug": "harmony-extracts-303-kush-live-wax",
      "variants": Array [],
    },
    "quantity": 0,
    "variantCount": 0,
    "zoneCount": 0,
  },
]
`;
